docs/reference/search/profile.asciidoc

```diff
@@ -1047,6 +1046,224 @@ per hit. In that case the true cost of `_source` phase is hidden in the
+{ "my-vector": [42, 8, -15] }
+{ "index": { "_id": "3" } }
+{ "my-vector": [15, 11, 23] }
+--------------------------------------------------
+
+With an index setup, we can now profile a kNN search query.
+
+[source,console]
+--------------------------------------------------
+POST my-knn-index/_search
+{
+  "profile": true, <1>
+  "knn": {
+    "field": "my-vector",
+    "query_vector": [-5, 9, -12],
+    "k": 3,
+    "num_candidates": 100
+  }
+}
+--------------------------------------------------
+// TEST[continued]
+
+<1> The `profile` parameter is set to `true`.
+
+In the response, we see a profile which includes a `knn` section
+as part of the `dfs` section for each shard along with profile output for the
+rest of the search phases.
+
+One of the `dfs.knn` sections for a shard looks like the following:
+
+[source,js]
+--------------------------------------------------
+"dfs" : {
+    "knn" : {
+        "query" : [
+            {
+                "type" : "DocAndScoreQuery",
+                "description" : "DocAndScore[100]",
+                "time_in_nanos" : 444414,
+                "breakdown" : {
+                  "set_min_competitive_score_count" : 0,
+                  "match_count" : 0,
+                  "shallow_advance_count" : 0,
+                  "set_min_competitive_score" : 0,
+                  "next_doc" : 1688,
+                  "match" : 0,
+                  "next_doc_count" : 3,
+                  "score_count" : 3,
+                  "compute_max_score_count" : 0,
+                  "compute_max_score" : 0,
+                  "advance" : 4153,
+                  "advance_count" : 1,
+                  "score" : 2099,
+                  "build_scorer_count" : 2,
+                  "create_weight" : 128879,
+                  "shallow_advance" : 0,
+                  "create_weight_count" : 1,
+                  "build_scorer" : 307595
+                }
+            }
+        ],
+        "rewrite_time" : 1275732,
+        "collector" : [
+            {
+                "name" : "SimpleTopScoreDocCollector",
+                "reason" : "search_top_hits",
+                "time_in_nanos" : 17163
+            }
+        ]
+    }
+}
+--------------------------------------------------
+// TESTRESPONSE[s/^/{\n"took": $body.took,\n"timed_out": $body.timed_out,\n"_shards": $body._shards,\n"hits": $body.hits,\n"profile": {\n"shards": [ {\n"id": "$body.$_path",\n/]
+// TESTRESPONSE[s/}$/}, "aggregations": [], "searches": $body.$_path, "fetch": $body.$_path}]}}/]
+// TESTRESPONSE[s/ (\-)?[0-9]+/ $body.$_path/]
+// TESTRESPONSE[s/"dfs" : \{/"dfs" : {"statistics": $body.$_path,/]
+
+In the `dfs.knn` portion of the response we can see the output
+the of timings for <<query-section, query>>, <<rewrite-section, rewrite>>,
+and <<collectors-section, collector>>. Unlike many other queries, kNN
+search does the bulk of the work during the query rewrite. This means
+rewrite_time represents the time spent on kNN search.
```

Comment:
Same comment here, we should say `rewrite_time`.

Reply:
Fixed.